pkg/cloudprovider/vm.go

```diff
@@ -22,6 +22,9 @@ import (
 type VMClient struct {
 	client   compute.VirtualMachinesClient
 	reporter *metrics.Reporter
+	// ARM throttling configures.
+	retryAfterReader time.Time
+	retryAfterWriter time.Time
```

Comment:
Are there going to be any race conditions for these two variables? If there are two 429 responses, the one that is being processed later will overwrite the `retryAfter*` time set by the first request.

Reply:
The `retryAfter` is per client which means we always want the latest possible retry after time. So if it gets overwritten to a later time by the second 429 response, then that is what we want to use.